src/core/public/application/application_service.tsx

```diff
@@ -105,23 +130,41 @@ export class ApplicationService {
     }
 
     // Disable registration of new applications
-    this.apps$.complete();
-    this.legacyApps$.complete();
+    this.started = true;
 
     const legacyMode = injectedMetadata.getLegacyMode();
     const currentAppId$ = new BehaviorSubject<string | undefined>(undefined);
-    const { availableApps, availableLegacyApps, capabilities } = await this.capabilities.start({
-      apps: new Map([...this.apps$.value].map(([id, { app }]) => [id, app])),
-      legacyApps: this.legacyApps$.value,
+    const { availableApps, capabilities } = await this.capabilities.start({
+      apps: new Map(
+        [...this.apps].map(([id, appBox]) => [
+          id,
+          (isLegacyApp(appBox) ? appBox : appBox.app) as App | LegacyApp,
+        ])
+      ),
       injectedMetadata,
     });
 
     // Only setup history if we're not in legacy mode
     const history = legacyMode ? null : createBrowserHistory({ basename: http.basePath.get() });
 
+    const availableApps$ = new BehaviorSubject<ReadonlyMap<string, App | LegacyApp>>(availableApps);
+
+    combineLatest([
+      of(availableApps),
+      this.statusUpdaters$.pipe(map(statusUpdaters => [...statusUpdaters.values()])),
+    ])
+      .pipe(
+        map(([apps, statusUpdaters]) => {
+          return new Map([...apps].map(([id, app]) => [id, updateStatus(app, statusUpdaters)]));
+        }),
+        map(apps => {
+          return new Map([...apps].filter(([id, app]) => app.status !== AppStatus.inaccessible));
+        })
+      )
+      .subscribe(apps => availableApps$.next(apps));
+
     return {
-      availableApps,
-      availableLegacyApps,
+      availableApps$,
```

Comment:
The separation between apps and legacy apps was kind of artificial here, and with the PR changes, having two distinct emitters for both was resulting in a lot of duplicate snippets. Also identification of both in the stream is trivial, so after discussion with @joshdover, I though it was better and more simple to merge them. (also this is only an internal API)